sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -140,6 +141,30 @@ def test_agglomerative_clustering_wrong_arg_memory():
     assert_raises(ValueError, clustering.fit, X)
 
 
+class Dummy(object):
```

Comment:
I think that those two class should be moved into the testing of `check_memory` and could be imported here.